pkl/version.go

```diff
@@ -0,0 +1,107 @@
+package pkl
+
+import (
+	"fmt"
+	"strconv"
+)
+
+//goland:noinspection GoSnakeCaseUsage
+var pklVersion0_25 = mustParseSemver("0.25.0")
+
+//goland:noinspection GoSnakeCaseUsage
+var pklVersion0_26 = mustParseSemver("0.26.0")
+
+type semver struct {
+	major      int
+	minor      int
+	patch      int
+	prerelease string
+	build      string
+}
+
+func compareInt(a, b int) int {
+	if a == b {
+		return 0
+	}
+	if a < b {
+		return -1
+	}
+	return 1
+}
+
+func (s *semver) compareToString(other string) int {
+	otherVersion, err := parseSemver(other)
+	if err != nil {
+		return 0
+	}
+	return s.compareTo(otherVersion)
+}
+
+// compareTo returns -1 if s < other, 1 if s > other, and 0 otherwise.
+func (s *semver) compareTo(other *semver) int {
+	comparison := compareInt(s.major, other.major)
+	if comparison != 0 {
+		return comparison
+	}
+	comparison = compareInt(s.minor, other.minor)
+	if comparison != 0 {
+		return comparison
+	}
+	// technically we should proceed to comparing prerelease versions, but we can skip
+	// this part because we don't have a use-case for it.
+	return compareInt(s.patch, other.patch)
```

Comment:
nit: we may not need it for now, but it wouldn't hurt to add a `strings.Compare()` for the `prerelease` and `build`, making the code future-proof.

Reply:
Good idea to compare prerelease too.

Build metadata should be excluded from comparison as per the semver spec: https://semver.org/#spec-item-10